### ISSUE TITLE
Feed de votações

### ISSUE THREAD

**Issue by leonardofl** [labhackercd/dados-abertos#31]
Proposta de funcionalidade: o sistema da Câmara poderia avisar sistemas registrados quando novas votações são adicionadas à base. Para isso algum sistema de feed teria que ser usado.

A principal motivação seria uma melhoria no desempenho e carga tanto do sistema da Câmara quanto dos sistemas clientes do WS, uma vez que os sistemas clientes não teriam que fazer mais tantas requisições para se manterem com uma cópia atualizada de todas as votações.

**Comment by ppKrauss** [comment]
+1 (boa  ideia!)

**Comment by CeninCD** [comment]
Bom dia, @leonardofl.

Concordo, seria uma boa para todos. Neste momento, em função de limitações de equipe e de outras demandas já priorizadas, não conseguimos avançar com essa implementação. 

Sugiro ao @labhacker  que a classifique como melhoria, para construirmos paulatinamente um backlog de novas funcionalidades aqui.

Outras contribuições nessa linha são bem vindas. Infelizmente, nossa capacidade de atendimento é restrita agora, mas estamos trabalhando para ampliá-la. E essas demandas nos ajudam a justificar.

Paulo Henrique 
Centro de Informática

**Comment by CeninCD** [comment]
Em tempo: existe um [método que listas as proposições votadas em plenário em um período](http://www2.camara.leg.br/transparencia/dados-abertos/dados-abertos-legislativo/webservices/proposicoes-1/ProposicoesVotadasEmPlenario). As aplicações de vocês podem utilizá-lo para obter as votações registradas depois da última atualização feita em suas bases de dados.

Paulo Henrique
Centro de Informática

**Comment by diraol** [comment]
Olá Paulo,

é isso que temos feito, mas a questão é que não sabemos "quando fazê-lo". O
Feed ajudaria a reduzir a quantidade de requests que faríamos no sistema de
vocês. Assim, ao invés de realizar requisições diárias ao método "listar
proposições" passando como parâmetro o dia de ontem, poderíamos fazer
apenas requisição quando houvesse atualização de fato nas proposições
inseridas no sistema. =)

E isso poderia valer para "novas proposições", mas poderia valer também
para proposições mais antigas, que já estavam no sistema, mas que sofreram
alguma alteração - situação que a consulta diária ao método "listar
proposições votadas em plenário" não consegue resolver.

**Comment by CeninCD** [comment]
Concordo, @diraol. O ideal seria o feed. Só complementei a resposta para mostrar uma solução de contorno com o que já temos.

Quanto às alterações nas proposições, há um outro [método que mostra as proposições tramitadas em um período](http://www2.camara.leg.br/transparencia/dados-abertos/dados-abertos-legislativo/webservices/proposicoes-1/listarProposicoesTramitadasNoPeriodo). Esse método não listaria alterações nos metadados das proposições, então não atende se o objetivo for acompanhar esse tipo de alteração. Embora alterações nos metadados sejam raras, podem acontecer.

São boas sugestões, vou inclui-las no nosso backlog também.

Paulo Henrique
Centro de Informática
